pkg/kubelet/cm/cpumanager/state/checkpoint.go

```diff
@@ -23,38 +23,87 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/checksum"
 )
 
-var _ checkpointmanager.Checkpoint = &CPUManagerCheckpoint{}
-
 // CPUManagerCheckpoint struct is used to store cpu/pod assignments in a checkpoint
-type CPUManagerCheckpoint struct {
+type CPUManagerCheckpoint = CPUManagerCheckpointV2
+
+var _ checkpointmanager.Checkpoint = &CPUManagerCheckpointV1{}
+var _ checkpointmanager.Checkpoint = &CPUManagerCheckpointV2{}
+
+// CPUManagerCheckpointV1 struct is used to store cpu/pod assignments in a checkpoint in v1 format
+type CPUManagerCheckpointV1 struct {
 	PolicyName    string            `json:"policyName"`
 	DefaultCPUSet string            `json:"defaultCpuSet"`
 	Entries       map[string]string `json:"entries,omitempty"`
 	Checksum      checksum.Checksum `json:"checksum"`
 }
 
+// CPUManagerCheckpointV2 struct is used to store cpu/pod assignments in a checkpoint in v2 format
+type CPUManagerCheckpointV2 struct {
+	PolicyName    string                       `json:"policyName"`
+	DefaultCPUSet string                       `json:"defaultCpuSet"`
+	Entries       map[string]map[string]string `json:"entries,omitempty"`
```

Comment:
nit: a map to a map can be cumbersome, we can come back to this later with an actual type.